examples/step-62/step-62.cc

```diff
@@ -1386,7 +1386,9 @@ int main(int argc, char *argv[])
 
       // Each of the simulations (displacement and calibration) is stored in a
       // separate HDF5 group:
-      for (const std::string &group_name : {"displacement", "calibration"})
+      const std::array<std::string, 2> group_names{
+        {"displacement", "calibration"}};
+      for (const std::string &group_name : group_names)
```

Comment:
Ah, that's a shame. What didn't work here? Would it have worked if you had made that
```
for (const char *group_name : {"displacement", "calibration"})
```

Reply:
I got a 'binding reference to a temporary' warning from GCC-13.

Using `char *`s won't work either since we do `group_name == displacement` on line 1421.